Add parent ID query (#8384)
opensearch-project/documentation-website: _query-dsl/joining/parent-id.md

---
layout: default
title: Parent ID
parent: Joining queries
nav_order: 40
---

# Parent ID query

> The `parent_id` query returns child documents whose parent document has the specified ID. You can establish parent-child relationships between documents in the same index by using a [join]({{site.url}}{{site.baseurl}}/field-types/supported-field-types/join/) field type.

Reviewer: ```suggestion
The `parent_id` query returns child documents whose parent documents have the specified ID. You can establish parent-child relationships between documents in the same index by using a [join]({{site.url}}{{site.baseurl}}/field-types/supported-field-types/join/) field type.
```

Author: The parent:child is a 1:many relationship so leaving singular.

> 

Reviewer: Above, global: We use "parent/child" (forward slash) above but "parent-child" here. Should be consistent globally.

## Example

Before you can run a `parent_id` query, your index must contain a [join]({{site.url}}{{site.baseurl}}/field-types/supported-field-types/join/) field in order to establish parent-child relationships. The index mapping request uses the following format:

```json
PUT /example_index
{
  "mappings": {
    "properties": {
      "relationship_field": {
        "type": "join",
        "relations": {
          "parent_doc": "child_doc"
        }
      }
    }
  }
}
```
{% include copy-curl.html %} 

For this example, first configure an index that contains documents representing products and their brands as described in the [`has_child` query example]({{site.url}}{{site.baseurl}}/query-dsl/joining/has-child/). 

> To search for children documents of a specific parent document, use a `parent_id` query. The following query returns child documents (products) whose parent document has the ID `1`:

Reviewer: ```suggestion
To search for child documents of a specific parent document, use a `parent_id` query. The following query returns child documents (products) whose parent document has the ID `1`:
```
